jujugui/static/gui/src/app/utils/environment-change-set.js

```diff
@@ -1365,18 +1357,75 @@ YUI.add('environment-change-set', function(Y) {
       return null;
     },
 
+    /**
+      Takes a service id and then handles unplacing all of the uncommitted
+      units for that service.
+
+      @method unplaceServiceUnits
+      @param {String} serviceId The service id to unplace units for.
+      @returns {Array} Any units that had been unplaced, or an empty array.
+    */
+    unplaceServiceUnits: function(serviceId) {
+      return this.get('db').units
```

Comment:
I used this approach because it's the same model list that we use when placing the units.